ComPact/Verifier/PactVerifier.cs

```diff
@@ -53,13 +53,32 @@ public async Task VerifyPactAsync(string path)
 
             if (pact.Interactions != null)
             {
-                if (_config.ProviderBaseUrl == null)
-                {
-                    throw new PactException("Could not verify pacts. Please configure a ProviderBaseUrl.");
+                if (_config.ProviderBaseUrl == null && _config.ProviderHttpClient == null)
+                {
+                    throw new PactException("Could not verify pacts. Please configure a ProviderBaseUrl or a pre-configured ProviderHttpClient.");
                 }
-                using (var client = new HttpClient())
-                {
-                    tests.AddRange(await VerifyInteractions(pact.Interactions, _config.ProviderBaseUrl, client.SendAsync, _config.ProviderStateHandler));
+
+                var existingClient = _config.ProviderHttpClient != null;
+
+                try
+                {
+                    if (!existingClient)
+                    {
+                        _config.ProviderHttpClient = new HttpClient();
+                    }
+
+                    tests.AddRange(await VerifyInteractions(pact.Interactions, _config.ProviderBaseUrl, _config.ProviderHttpClient.SendAsync, _config.ProviderStateHandler));
```

Comment:
What do you think, should the ProviderBaseUrl be leading even if the ProviderHttpClient already contains a BaseAddress?

Reply:
Yeah so I wondered about this when I was writing the PR. Following the code through though, ProviderBaseUrl is only used in Request.ToHttpRequestMessage line 63:
`var uri = baseUrl + Path;`

I wasn't sure if there was any scenario where this would still be valid even with a pre-configured HttpClient since the baseUrl just gets prepended to the path?

Reply:
Yeah this is a bit stange, but it's because HttpRequestMessage is awkward to work with. I previously implemented this using RestSharp, which was much nicer, but I wanted to get rid of the dependency.
It's a bit ugly, but something like this might do the trick: 
`var baseUrl = _config.ProviderHttpClient?.BaseAddress.ToString() ?? _config.ProviderBaseUrl;`

Reply:
If we do that, the baseUrl will be applied twice since it will be the base address in HttpClient, but then also pre-pended to the request.

I think this is ok as-is _unless_ we say that you can't the baseUrl is not used if HttpClient is passed in? I can't decide if there would ever be a valid case to provide both?

Reply:
I just tested this a bit and it seems like we can actually drop `baseUrl` as parameter from `Request.ToHttpRequestMessage()`.
You could then just do: `var client = _config.ProviderHttpClient ??  new HttpClient { BaseAddress = new Uri(_config.ProviderBaseUrl) };`

Reply:
Ah yes! That's a much nicer and cleaner way to do it!